src/devices_models/device_constructors/thermalgeneration_constructor.jl

```diff
@@ -2,18 +2,18 @@ function construct_device!(
     container::OptimizationContainer,
     sys::PSY.System,
     ::ArgumentConstructStage,
-    ::DeviceModel{T, FixedOutput},
-    network_model::NetworkModel{S},
-) where {T <: PSY.ThermalGen, S <: PM.AbstractActivePowerModel}
+    device_model::DeviceModel{T, FixedOutput},
+    network_model::NetworkModel{<:PM.AbstractActivePowerModel},
+) where {T <: PSY.ThermalGen}
     devices =
-        get_available_components(T, sys, get_attribute(model, "filter_function"))
-    add_parameters!(container, ActivePowerTimeSeriesParameter, devices, model)
+        get_available_components(T, sys, get_attribute(device_model, "filter_function"))
+    add_parameters!(container, ActivePowerTimeSeriesParameter, devices, device_model)
```

Comment:
This was an existing bug because `model` was not defined. Please confirm the fix.